ignite/handlers/lr_finder.py

```diff
@@ -118,13 +118,25 @@ def _run(
             )
 
         self.logger.debug(f"Running LR finder for {num_iter} iterations")
-        if start_lr is None:
-            start_lr = optimizer.param_groups[0]["lr"]
         # Initialize the proper learning rate policy
-        if step_mode.lower() == "exp":
-            start_lr_list = [start_lr] * len(optimizer.param_groups)
-            self._lr_schedule = LRScheduler(_ExponentialLR(optimizer, start_lr_list, end_lr, num_iter))
+        if step_mode.lower() == "exp":       
+            if start_lr is None:
+                start_lr_list = [optimizer.param_groups[i]["lr"] for i in range(len(optimizer.param_groups))]
+            elif isinstance(start_lr, float):
```

Comment:
We need to update start_lr and end_lr type hints in this method and in `attach` to relfect that we can pass: float or List[float].

Reply:
`elif isinstance(start_lr, list[float]):
                start_lr_list = []
                for value in start_lr:
                    temp_start_lr = value * len(optimizer.param_groups)
                    start_lr_list.append(temp_start_lr)`
Can I perform it like that?

Reply:
Type hints are about decorating args and functions. Here is `_run` function:
```python
    def _run(
        self,
        trainer: Engine,
        optimizer: Optimizer,
        output_transform: Callable,
        num_iter: int,
        start_lr: float,    # <---- this is a type hint for start_lr
        end_lr: float,
        step_mode: str,
        smooth_f: float,
        diverge_th: float,
) -> None:
```
so, we see that `start_lr` is of type float. But as in code we also accept a list of floats and also None, we have to update start_lr type hint:
```
start_lr: Union[None, float, List[float]]
```

Check and update also `attach` method.

Reply:
> `elif isinstance(start_lr, list[float]): start_lr_list = [] for value in start_lr: temp_start_lr = value * len(optimizer.param_groups) start_lr_list.append(temp_start_lr)` Can I perform it like that?

@Jacob208M , if you want to have a pretty code block in your comment, you can enclose it between two lines of \```. If the language of the block is a specific one e.g. Python, you can write its name just after the first ```.

example:
````
```python
# some code
a = 3
```
````

becomes:
```python
# some code
a = 3
```
```